packages/clerk-js/src/utils/captcha/turnstile.ts

```diff
@@ -112,59 +116,71 @@ async function loadCaptchaFromFAPIProxiedURL(fallbackUrl: string) {
  * - If the widgetType is 'smart', the captcha widget is rendered in a div with the id 'clerk-captcha'. If the div does
  *  not exist, the invisibleSiteKey is used as a fallback and the widget is rendered in a hidden div at the bottom of the body.
  */
-export const getTunstileToken = async (captchaOptions: {
-  siteKey: string;
-  scriptUrl: string;
-  widgetType: CaptchaWidgetType;
-  invisibleSiteKey: string;
-}) => {
-  const { siteKey, scriptUrl, widgetType, invisibleSiteKey } = captchaOptions;
-  let captchaToken = '',
-    id = '';
-  let isInvisibleWidget = !widgetType || widgetType === 'invisible';
+export const getTurnstileToken = async (opts: CaptchaOptions) => {
+  const { siteKey, scriptUrl, widgetType, invisibleSiteKey } = opts;
+  const { modalContainerQuerySelector, modalWrapperQuerySelector, closeModal, openModal } = opts;
+  const captcha: Turnstile = await loadCaptcha(scriptUrl);
+  const errorCodes: (string | number)[] = [];
+
+  let captchaToken = '';
+  let id = '';
   let turnstileSiteKey = siteKey;
+  let retries = 0;
+  let widgetContainerQuerySelector: string | undefined;
+  const widgetTypeUsed = [widgetType || ''];
```

Comment:
I don't understand why this is an array and we combine these at the end. I think that the backend expects `smart` or `invisible` here as a value.

Reply:
We want to know if the challenge got escalated from invisible/non-interactive to interactive

Reply:
Until now, we used this parameter to hint backend in order to know which of the two site-keys was actually used, so that it can use the respective one in the backend to verify the token with Cloudflare.